package/src/bin/omicron-package.rs

```diff
@@ -926,6 +926,62 @@ impl Config {
             _ => bail!("Aborting"),
         }
     }
+
+    /// Returns target packages to be assembled on the builder machine, limited
+    /// to those specified in `only` (if set).
+    // TODO: should this be moved to omicron-zone-package?
+    fn packages_to_build(&self) -> PackageMap<'_> {
+        let packages = self.package_config.packages_to_build(&self.target);
+        if self.only.is_empty() {
+            packages
+        } else {
+            // create a new PackageMap and copy over the selected packages
+            let mut filtered_packages = PackageMap(BTreeMap::new());
+            for package_name in &self.only {
```

Comment:
Looks like this will silently continue if someone typos the name of a `--only` package. Do you think it would be better to throw an explicit error in this case?